services/geocoding.js

```diff
@@ -0,0 +1,101 @@
+'use strict';
+
+var v = require('./service-helpers/validator');
+var pick = require('./service-helpers/pick');
+var createServiceFactory = require('./service-helpers/create-service-factory');
+
+/**
+ * Geocoding API service.
+ */
+var Geocoding = {};
+
+/**
+ * Search for a place.
+ *
+ * See the [public documentation](https://www.mapbox.com/api-documentation/#search-for-places).
+ *
+ * @param {Object} config
+ * @param {string} config.query - A place name.
+ * @param {'mapbox.places'|'mapbox.places-permanent'} config.mode
+ * @param {string|Array<string>} [config.country]
+ * @param {[number, number]} [config.proximity] - `[longitude, latitude]`
+ * @param {Array<string>} [config.types]
+ * @param {boolean} [config.autocomplete=true]
+ * @param {[number, number, number, number]} [config.bbox] - `[minX, minY, maxX, maxY]`
+ * @param {number} [config.limit=5]
+ * @param {string|Array<string>} [config.language]
+ * @return {MapiRequest}
+ */
+Geocoding.forwardGeocode = function(config) {
+  v.assertShape({
+    query: v.required(v.string),
+    mode: v.required(v.oneOf('mapbox.places', 'mapbox.places-permanent')),
+    country: v.oneOfType(v.string, v.arrayOf(v.string)),
+    proximity: v.coordinates,
+    types: v.arrayOf(v.string),
```

Comment:
We could check for all options here `Options are  country ,  region ,  postcode ,  district , place ,  locality ,  neighborhood ,  address ,  poi , and  poi.landmark .`